src/main/java/com/teragrep/cfe_39/consumers/kafka/HDFSWrite.java

```diff
@@ -0,0 +1,250 @@
+                    // Create new Directory
+                    fs.mkdirs(newDirectoryPath);
+                    LOGGER.debug("Path <{}> created.", path);
+                }
+
+                //==== Write file
+                LOGGER.debug("Begin Write file into hdfs");
+                //Create a path
+                Path hdfswritepath = new Path(newDirectoryPath + "/" + fileName); // filename should be set according to the requirements: 0.12345 where 0 is Kafka partition and 12345 is Kafka offset.
+                if (fs.exists(hdfswritepath)) {
+                    throw new RuntimeException("File " + fileName + " already exists");
+                }
+
+                //Init output stream
+                FSDataOutputStream outputStream = fs.create(hdfswritepath);
+                // Write the file contents of syslogFile to hdfswritepath in HDFS.
+                // file to bytes[]
+
+                /*byte[] bytearray = new byte[(int) syslogFile.length()];
+                try (FileInputStream inputStream = new FileInputStream(syslogFile)) {
+                    inputStream.read(bytearray);
+                }*/
+                byte[] bytes = Files.readAllBytes(Paths.get(syslogFile.getPath()));
+                outputStream.write(bytes);
+
+                outputStream.close();
+                LOGGER.debug("End Write file into hdfs");
+                boolean delete = syslogFile.delete(); // deletes the avro-file from the local disk now that it has been committed to HDFS.
+                LOGGER.debug("\nFile committed to HDFS, file writepath should be: <{}>\n", hdfswritepath);
+
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void updateTimestamp(Path hdfswritepath, long lastEpochMicros) {
+        // Testing timestamp editing. The new timestamp should be the timestamp of the last record that was added to the AVRO-file.
+        try {
+            FileSystem fs_temp = FileSystem.get(URI.create(hdfsuri), conf);
+            FSDataOutputStream fsDataOutputStream = fs_temp.create(hdfswritepath);
+            fs_temp.setTimes(hdfswritepath, lastEpochMicros, -1); // where mtime is modification time and atime is access time. -1 as input parameter leaves the original atime/mtime value as is.
+            fsDataOutputStream.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // try-with-resources handles closing the filesystem automatically.
+    public void close() {
+        // FIXME: fs.close() doesn't just affect the current class, it affects all the FileSystem objects that were created using FileSystem.get(URI.create(hdfsuri), conf); in different threads.
+        /*if (fs != null) {
+            try {
+                fs.close();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }*/
```

Comment:
FIXME / Don't comment out code

Reply:
Solved in commit 022832c8b423efc7466b7aead56cc317a45a0002